### ISSUE TITLE
Score Export Broken

### ISSUE THREAD

**Issue by Garfield96** [ls1intum/Artemis#4815]
### Describe the bug

The points for individual programming exercises in the CSV file exported via `Scores` do not match the displayed points in the UI. For example, we had a student with 100% in the web UI, but according to the CSV, he/she didn't even participate in the exercise. In addition, the sum of exported points does not match `Programming Points` in the CSV. However, `Programming Points` is equal to the web UI.

We assume, that programming points are correctly calculated, but the export of points for individual programming exercises is broken.

Incorrect output:
| Name | Username | ... | Binary Search Tree | Bash | ASCII Tabelle | ... | Programming Points | ... |
| --- | --- | --- | --- | --- | --- | --- | --- | --- |
| xyz | xyz | ... | `<wrong points>` | `<wrong points>` | `<wrong points>` | ... | `<sum>` | ... |

### To Reproduce

1. Go to `Course Management`
2. Select a course with several programming exercises
3. Click on `Scores`
4. Click on `Export as CSV`
5.1 Check whether the sum of points received in individual programming exercises match `Programming Points` 
5.2 Make sure that points attributed to an exercise belong to this exercise.

### Expected behavior

1. Sum of individual programming exercises matches `Programming points`.
2. Points for individual programming exercises belong to the exercise.

### What browsers are you seeing the problem on?

Firefox

**Comment by b-fein** [comment]
Artemis 5.6.1 with the partial fix from #4816 where the points did not match the exercises has been released. Has this also fixed the points sum issue?

**Comment by Garfield96** [comment]
Unfortunately, the fix has broken the whole `Scores` page. The webpage is unresponsive and clicking on "Export as CSV" results in the following error message:
```
ERROR TypeError: r.user.name is undefined
    setUserInformation course-scores-csv-row-builder.ts:70
    generateStudentStatisticsCsvRow course-scores.component.ts:601
    exportResults course-scores.component.ts:523
    exportResults course-scores.component.ts:523
    Ih course-scores.component.html:87
    Angular 3
    U helpers.js:73
    Angular 15
    Eo trycatch.js:183
    Zo instrument.js:449
    Angular 6
    Ih course-scores.component.html:87
    Angular 9
    Hh course-scores.component.html:87
    Angular 6
    highlightBar course-scores.component.ts:912
    calculateCourseStatistics course-scores.component.ts:259
    RxJS 28
    Vt Angular
    U helpers.js:73
    Angular 15
    Eo trycatch.js:183
    Zo instrument.js:458
    handle Angular
    RxJS 20
instrument.js:109:42
```

**Comment by b-fein** [comment]
Okay, the export at the moment assumes that every user has a name, which apparently is not necessary. That is an easy fix. I’ll make a PR for that.

**Comment by Garfield96** [comment]
Yes, there is a student without a name and email address.

**Comment by krusche** [comment]
Can you send me the student without a name via Slack? Then I can check why this is the case and if this is an error on the database...

**Comment by Garfield96** [comment]
> Artemis 5.6.1 with the partial fix from #4816 for the case where the points did not match the exercises has been released. Has this also fixed the points sum issue?

The student name was now manually assigned in the database, and the export is working again. The sums are now correct. Thus this issue can be closed after #4860 was merged.
